integration-tests/tests/src/tests/array-buffer.ts

```diff
@@ -0,0 +1,219 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Buffer } from "buffer";
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+const SingleSchema: Realm.ObjectSchema = {
+  name: "PrimitiveData",
+  properties: {
+    a: "data",
+  },
+};
+
+type ISingleSchema = {
+  a: ArrayBuffer;
+};
+
+describe("ArrayBuffer type", () => {
+  openRealmBeforeEach({ schema: [SingleSchema] });
+  it("supports storing data from a bufferView", function (this: RealmContext) {
+    const view = new Uint8Array([
+      0xd8,
+      0x21,
+      0xd6,
+      0xe8,
+      0x00,
+      0x57,
+      0xbc,
+      0xb2,
+      0x6a,
+      0x15,
+      0x77,
+      0x30,
+      0xac,
+      0x77,
+      0x96,
+      0xd9,
+      0x67,
+      0x1e,
+      0x40,
+      0xa7,
+      0x6d,
+      0x52,
+      0x83,
+      0xda,
+      0x07,
+      0x29,
+      0x9c,
+      0x70,
+      0x38,
+      0x48,
+      0x4e,
+      0xff,
+    ]);
+
+    this.realm.write(() => this.realm.create(SingleSchema.name, { a: view }));
+
+    const data = this.realm.objects<ISingleSchema>(SingleSchema.name)[0];
+    expect(data.a.byteLength).equals(view.byteLength, "Data size should be equal");
+
+    const binary_view = new Uint8Array(data.a);
+
+    for (let i = 0; i < view.byteLength; i++) {
+      const p1 = view[i];
+      const p2 = binary_view[i];
+      expect(p1).equals(p2, "buffers should be equal");
+    }
+  });
+  it("supports storing data from a bufferDataView", function (this: RealmContext) {
+    const view = new Uint8Array([
+      0xd8,
+      0x21,
+      0xd6,
+      0xe8,
+      0x00,
+      0x57,
+      0xbc,
+      0xb2,
+      0x6a,
+      0x15,
+      0x77,
+      0x30,
+      0xac,
+      0x77,
+      0x96,
+      0xd9,
+      0x67,
+      0x1e,
+      0x40,
+      0xa7,
+      0x6d,
+      0x52,
+      0x83,
+      0xda,
+      0x07,
+      0x29,
+      0x9c,
+      0x70,
+      0x38,
+      0x48,
+      0x4e,
+      0xff,
+    ]);
+
+    this.realm.write(() => this.realm.create(SingleSchema.name, { a: new DataView(view.buffer) }));
+
+    const data = this.realm.objects<ISingleSchema>(SingleSchema.name)[0];
+    expect(data.a.byteLength).equals(view.byteLength, "Data size should be equal");
+
+    const binary_view = new Uint8Array(data.a);
+
+    for (let i = 0; i < view.byteLength; i++) {
+      const p1 = view[i];
+      const p2 = binary_view[i];
+
+      expect(p1).equals(p2, "buffers should be equal");
+    }
+  });
+  it("supports storing data from an arrayBuffer", function (this: RealmContext) {
```

Comment:
Since this is referring to an actual type: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer

```suggestion
  it("supports storing data from an ArrayBuffer", function (this: RealmContext) {
```

It could be updated in more places below too.